src/main/java/org/jabref/gui/undo/RedoAction.java

```diff
@@ -21,11 +25,15 @@ public RedoAction(Supplier<LibraryTab> tabSupplier, DialogService dialogService,
         this.tabSupplier = tabSupplier;
         this.dialogService = dialogService;
 
-        // TODO: The old listener should be removed. Otherwise, memory consumption will increase.
-        stateManager.activeTabProperty().addListener((observable, oldValue, activeLibraryTab) -> {
+        ChangeListener<Optional<LibraryTab>> listener = (observable, oldValue, activeLibraryTab) -> {
             activeLibraryTab.ifPresent(libraryTab ->
                     this.executable.bind(libraryTab.getUndoManager().getRedoableProperty()));
-        });
+
+            oldValue.ifPresent(libraryTab -> this.executable.unbind());
```

Comment:
Thinking a bit more, I am a bit concerned about the order of operations here.
Case 1:
State A->B - change detected, bind to B, unbind if oldValue is not null. After this,
State B->A - change detected, bind to A (not needed), unbind if B (oldvalue) is not null, which is true, so unbind (this is fine).

Case 2:
State A->B - change detected, bind to B, unbind if oldValue is not null. After this,
State B->C - change detected, bind to C, unbind if B is not null (which is true) - thus executable is not bound.

@Siedlerchr would it be better if we don't unbind at all here? Or is this alright as LibraryTab can never have a third state "C"?